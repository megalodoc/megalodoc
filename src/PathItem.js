import React from 'react'
import Operation from './Operation'
import PropTypes from 'prop-types'

const PathItem = ({pathName, pathItem}) =>
  <div>
    <h2>{pathName}</h2>
    {
      Object.keys(pathItem).map(oKey =>
        pathItem[oKey]
        ? <Operation key={oKey} verb={oKey} operation={pathItem[oKey]} />
        : null
      )
    }
  </div>

PathItem.propTypes = {
  pathName: PropTypes.string.isRequired,
  pathItem: PropTypes.shape({
    get: PropTypes.object,
    put: PropTypes.object,
    post: PropTypes.object,
    delete: PropTypes.object,
    parameters: PropTypes.array
  }).isRequired
}

export default PathItem
