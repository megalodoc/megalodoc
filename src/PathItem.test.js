import React from 'react'
import renderer from 'react-test-renderer'
import PathItem from './PathItem'
import petstore from './fixtures/petstore.json'

it('renders the PathItem component', () => {
  const paths = petstore.paths
  const firstPath = Object.keys(paths)[0]
  const component = renderer.create(<PathItem pathName={firstPath} pathItem={paths[firstPath]} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
