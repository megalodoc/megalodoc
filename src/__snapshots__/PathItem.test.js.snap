// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PathItem component 1`] = `
<div>
  <h2>
    /pet
  </h2>
  <div>
    <h3>
      post
    </h3>
    <p>
      Add a new pet to the store
    </p>
    <p>
      
    </p>
  </div>
  <div>
    <h3>
      put
    </h3>
    <p>
      Update an existing pet
    </p>
    <p>
      
    </p>
  </div>
</div>
`;
